### PR TITLE
Feat: Testing and Linting

### PR DESCRIPTION
**Pull request by CodeWithEmad**
Assorted testing and linting improvements:

* Makefile with test and lint targets
* Long lines reformated with black
* Sort imports with isort

In addition:
* Typing added for various functions
* All configs moved to 1 place

### CODE REVIEW

**Comment by FahadKhalid210** [comment]
Hi @CodeWithEmad Thanks for the PR. I will look into it.
Can you please resolve the conflicts?

**Comment by CodeWithEmad** [comment]
Done.

**Review by FahadKhalid210 — APPROVED** [review]
LGTM 👍

File: tutorcairn/plugin.py [hunk 1]
```diff
@@ -18,51 +18,53 @@
...
-        ("CAIRN_POSTGRESQL_USERNAME", "superset"),
-        (
-            "CAIRN_SUPERSET_DOCKER_IMAGE",
-            "{{ DOCKER_REGISTRY }}overhangio/cairn-superset:{{ CAIRN_VERSION }}",
-        ),
-        ("CAIRN_SUPERSET_LANGUAGE_CODE", "{{ LANGUAGE_CODE[:2] }}"),
+        "RUN_POSTGRESQL": True,
+        "POSTGRESQL_DATABASE": "superset",
+        "POSTGRESQL_USERNAME": "superset",
+        "SUPERSET_DOCKER_IMAGE": "{: DOCKER_REGISTRY }}overhangio/cairn-superset:{{ CAIRN_VERSION }}",
```
Comment (regisb) [inline tutorcairn/plugin.py:39]: FYI there is a typo that is breaking the build here: `{: DOCKER_REGISTRY }}` -> `{{ DOCKER_REGISTRY }}`
I'm pushing a change now.

**Pull request merged**